dcos/dcos-ui
# DCOS-21472: Delete last endpoint from the form

Closes DCOS-21472

## Testing

1. Launch the app (it's a Multicontainer app, or a Pod in other terminology)
2. Go to the edit form
3. Open JSON editor to watch JSON changes
4. On the networking tab delete the endpoint
5. Assert it should be gone from the JSON

```
{
  "id": "/last-endpoint",
  "containers": [
    {
      "name": "container-1",
      "resources": {
        "cpus": 0.001,
        "mem": 32,
        "disk": 0
      },
      "image": {
        "kind": "DOCKER",
        "id": "nginx"
      },
      "endpoints": [
        {
          "name": "asd",
          "hostPort": 9999,
          "protocol": [
            "tcp"
          ]
        }
      ]
    }
  ],
  "networks": [
    {
      "mode": "host"
    }
  ],
  "scaling": {
    "instances": 1,
    "kind": "fixed"
  },
  "scheduling": {
    "placement": {
      "constraints": []
    }
  },
  "executorResources": {
    "cpus": 0.1,
    "mem": 32,
    "disk": 10
  },
  "volumes": [],
  "fetch": []
}
```

## Screenshots

Bug in the nutshell:

![delete-endpoint](https://user-images.githubusercontent.com/186223/54680725-e8dc0400-4b0a-11e9-8892-1c7e6f986b7a.gif)
